src/main/java/uk/gov/pay/webhooks/deliveryqueue/managed/WebhookMessagePollingService.java

```diff
@@ -0,0 +1,55 @@
+package uk.gov.pay.webhooks.deliveryqueue.managed;
+
+import org.hibernate.SessionFactory;
+import org.hibernate.context.internal.ManagedSessionContext;
+import uk.gov.pay.webhooks.deliveryqueue.dao.WebhookDeliveryQueueDao;
+import uk.gov.pay.webhooks.deliveryqueue.dao.WebhookDeliveryQueueEntity;
+
+import javax.inject.Inject;
+import java.util.Optional;
+
+public class WebhookMessagePollingService {
+    private final WebhookDeliveryQueueDao webhookDeliveryQueueDao;
+
+    private final SendAttempter sendAttempter;
+
+    private final SessionFactory sessionFactory;
+
+    @Inject
+    public WebhookMessagePollingService(WebhookDeliveryQueueDao webhookDeliveryQueueDao, SendAttempter sendAttempter, SessionFactory sessionFactory) {
+        this.webhookDeliveryQueueDao = webhookDeliveryQueueDao;
+        this.sendAttempter = sendAttempter;
+        this.sessionFactory = sessionFactory;
+    }
+
+    public void pollWebhookMessageQueue() {
+        Optional<WebhookDeliveryQueueEntity> attemptCursor;
+
+        do {
+            attemptCursor = sendIfAvailable();
+        } while(attemptCursor.isPresent());
```

Comment:
I wonder if it might just be simpler for `sendIfAvailable()` to return a boolean (since we never seem to look at anything inside the `WebhookDeliveryQueueEntity`).